apollo-api/src/main/java/com/spotify/apollo/RequestMetadata.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * -\-\-
+ * Spotify Apollo API Interfaces
+ * --
+ * Copyright (C) 2013 - 2016 Spotify AB
+ * --
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * -/-/-
+ */
+package com.spotify.apollo;
+
+import java.util.Optional;
+
+/**
+ * Describes an API for retrieving metadata about an incoming request.
+ */
+public interface RequestMetadata {
+
+  /**
+   * Returns the class that is the source of the metadata; this enables users to more easily
+   * understand where the metadata comes from and how to interpret data like the {@link #protocol()}.
+   */
+  Class<?> source();
```

Comment:
Hmm... With the current set of metadata fields, we will only need to add them in one place. With the solution in #167, that has a general map of metadata properties, there might be value in allowing 'appending' to the metadata properties in more than one place.

The reason for using a `Class<?>` rather than a `String` as the source name is to facilitate tracking down where the metadata was added. It makes it harder for implementors to provide a bad/hard to find source.